File: src/com/prey/PreyLogger.java
/*******************************************************************************
 * Created by Carlos Yaconi
 * Copyright 2012 Fork Ltd. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey;

import android.util.Log;

public class PreyLogger {

	public static void d(String message) {
		//if (PreyConfig.LOG_DEBUG_ENABLED)
			Log.i(PreyConfig.TAG,message);
	}

	public static void i(String message) {
		Log.i(PreyConfig.TAG,message);
	}

	public static void e(final String message, Throwable e) {
		if (e!=null)
			Log.e(PreyConfig.TAG, message, e);
		else
			Log.e(PreyConfig.TAG, message);
	}

}

File: app/src/main/java/com/prey/PreyLogger.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey

import android.content.Context
import android.util.Log

class PreyLogger {

    fun d(message: String?) {
        Log.i(com.prey.PreyConfig.TAG, "OSO: $message")
    }

    fun i(message: String?) {
        Log.i(PreyConfig.TAG, message!!)
    }

    fun e(message: String?, e: Throwable?) {
        if (e != null) Log.e(PreyConfig.TAG, message, e)
        else Log.e(PreyConfig.TAG, message!!)
    }
    companion object {

        private var instance: PreyLogger? = null
        fun getInstance(): PreyLogger {
            if (instance == null) {
                instance = PreyLogger()
            }
            return instance!!
        }

        fun i(message: String?) {
            Log.i(PreyConfig.TAG, "OSO_:$message")
        }
        fun d(message: String?) {
            Log.i(PreyConfig.TAG, "OSO_:$message")
        }
        fun e(message: String?, e: Throwable?) {
            Log.e(PreyConfig.TAG, "OSO_:$message", e)
        }
    }


}